.github/workflows/build-and-test.yml

```diff
@@ -208,37 +150,73 @@ jobs:
           gem push pkg/*.gem
           rm -f $HOME/.gem/credentials
 
+  compute-tag:
+    runs-on: ubuntu-latest
+    outputs:
+      BRANCH_NAME: ${{ steps.branch_env.outputs.branch_name }}
+    steps:
+      - name: Set branch environment variables
+        id: branch_env
+        run: |
+          #BRANCH_NAME=${GITHUB_REF#refs/heads/}
+          BRANCH_NAME=${{ github.head_ref || github.ref_name }}
+          echo "branch_name=$BRANCH_NAME" >> $GITHUB_OUTPUT
+          echo "is_master=${{ github.ref == 'refs/heads/master' }}" >> $GITHUB_OUTPUT
+
+  whats-my-tag:
+    needs: compute-tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: 
+        run: |
+          echo "branch name=${{ needs.compute-tag.outputs.BRANCH_NAME }}"
+          echo "image=${{ format('ghcr.io/{0}/fig-ci:centos7.9.2009-{1}', github.repository, needs.compute-tag.outputs.BRANCH_NAME) }}"
+          
 # Need to figure out how to parameterize the common parts of
 # these jobs and factor out only the differing parts.
   test-gem-installation-centos:
-    if: false
-    needs: package
+    # Enable the job
+    needs: [ package, compute-tag]
     runs-on: ubuntu-latest
     container:
-      image: centos:7.9.2009
+      # Use stable tag on master, branch-based tag on feature branches
+      image: ${{ github.ref == 'refs/heads/master' && format('ghcr.io/{0}/fig-ci:centos7.9.2009', github.repository) || format('ghcr.io/{0}/fig-ci:centos7.9.2009-{1}', github.repository, needs.compute-tag.outputs.BRANCH_NAME) }}
+      options: --user root
+
     strategy:
       matrix:
-        ruby:
-          - 3.1.2
-          - 3.3
+        ruby-version:
+          - "3.1.2"
+          - "3.3.7"
     steps:
-      - name: Download gem
-        uses: actions/download-artifact@v4
-        with:
-          name: fig
-          path: .
-
-      - name: Install packages required by setup-ruby action
+      - name: Install gh cli
         run: |
-          yum install -y libyaml
+          curl -s https://cli.github.com/packages/rpm/gh-cli.repo -o /etc/yum.repos.d/gh-cli.repo && yum install -y gh
```

Comment:
Are we running these on our own runners? I thought the GH CLI was [installed](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-github-cli-in-workflows) on the GitHub provided runners?

Reply:
This runs on github's runner, and ghcli is installed on their runner images.  HOWEVER, *this* job is running in a CONTAINER on their runner using the centos 7.9 image curated in the docker-image workflow, so that container doesn't have gh.

The reason I have gh in here is because I want access to an artifact created in a previous job. Normally one would use `actions/download-artifact`, but the supported `v4` version of that action requires node20 and that's not available as a binary in centos 7.9.  The `v3` version would work, but it's no longer available as a standard action. I thought about vendoring it for our project, but that seemed like both a bigger immediate hurdle as a much bigger long-term maintenance nightmare than just using gh.